### PR TITLE
 The Syndicate MODs are now resistant to batons knockdown.

### PR DESCRIPTION
**Pull request by EnterTheJake**
## About The Pull Request

Adds a new Module preinstalled into the nukie suits, and purchasable in the uplink,  grants resistance to batons knockdown.

## Why It's Good For The Game

At the last coderbus meeting, i personally asked to Fikou and Mothblocks for more baton counters to be introduced, they seemed to be ok with the idea, so here we are.

Introduces an unremovable module which grants to the wearer resistance to baton knockdown, it comes preinstalled in all syndicate modsuits.

With the fix of "rest to prevent disarm", stun batons have gotten even more powerful than they once were, and while i do agree secret kinda abusable tech is bad for the health of the game, the end result is that the stun meta is even more prominent than ever, so it's a good time introduce a small counter to it.

From a thematic perspective, Syndie suits should at least make people feel afraid, which is hard to do when a single prod from a stun stick will make you drop like a sack of potatoes, lose your desword and get slashed to bits.

On the gameplay side, wearing one of these will make you a target for the entire station, if a traitor is willing to expose themselves, or in the case of nukies, forced to; they should probably have a small edge on what's the most common tool used to stop them.

Keep in mind that this only prevents the knockdown, it does nothing for  the stamina damage, you'll still get stamcritted in a couple of hits, nothing changes on that front.

This is merely to prevent a scenario where a single baton hit will immediately lose you a fight.

## Changelog

:cl:
add: adds the MOD shock-absorption module, into the game.
add: The MOD shock-absorption module into the the uplinks, costs 4 TC.
balance: Nukie modsuits come with the shock_absorption module preinstalled.
/:cl:

### CODE REVIEW

**Review by Fikou — CHANGES_REQUESTED** [review]
i want this to be removable, take complexity/power and have a visible sprite on the mob, it should not be inbuilt to the infiltrator theme

**Comment by EnterTheJake** [comment]
> i want this to be removable, take complexity/power and have a visible sprite on the mob, it should not be inbuilt to the infiltrator theme

should i also add this to the uplink?

**Comment by EnterTheJake** [comment]
> if you want to. you can also add it to black market and whatever, i just dont like the recent trend of invisible module that affects combat where you just need to know that this specific suit has one- it bites on the modularity of the system when everything is inherent the sprite can be whatever you think would work to visually convey it, some sort of pad on the shoulders, thing attached to belt, etc

I assume just getting rid of the module and baking it into the suit themselves is out of the question.

**Comment by Higgin** [comment]
I can't remotely stress how tweaked this will be in the hands of ops and anyone with access to a little bit of extra prep. Free knockdown immunity, in no-grav/jetpack settings especially, is going to be ridiculous for how hard it makes to disarm or contest people.

And then, stack that with shields and the inherent block chance on many Syndie weapons. 

This shouldn't come within a ten foot pole of the current balance. The base MOD can also be found in space and rarely from LL Legion mobs.

Syndies (granted - not spies that just happen to get one) already have access to stimpacks, which are a time-sensitive way of getting these effects and more with a cost and a counter (literally just shoot them, wait, or dragnet them - some combo thereof - two of which might work here for something a lot of antags get for free.)

Honestly, I'd feel a lot better about this if it was just a two-time stimpack autoinjector that triggers on hitting softcrit or manually.

This is just a crazy buff across the board around how people in Syndie mods usually go down, but it won't make everyone jump the hardsuit guy in red - it'll make everyone who has to deal with hardsuit guy as a restricted antag on MRP think they do and shoot him in the middle of a soliloquy to steal his drip (basically treating him more like a desword nukie even if he's not.)

This ain't all bad, because fictionally - wearing the stuff and having it be scary is cool if that's what you're after, and why would you wear the tech of the terrible Cybersun/Gorlex killsquads if you didn't absolutely have to?

I'm just concerned at how much of a swing on balance this is going to be - baton knockdown is unfortunately a cornerstone fallback even as we chip away at stun combat.

**Comment by EnterTheJake** [comment]
> I can't remotely stress how tweaked this will be in the hands of ops and anyone with access to a little bit of extra prep. Free knockdown immunity, in no-grav/jetpack settings especially, is going to be ridiculous for how hard it makes to disarm or contest people.
> 
> And then, stack that with shields and the inherent block chance on many Syndie weapons.
> 
> This shouldn't come within a ten foot pole of the current balance. The base MOD can also be found in space and rarely from LL Legion mobs.
> 
> Syndies (granted - not spies that just happen to get one) already have access to stimpacks, which are a time-sensitive way of getting these effects and more with a cost and a counter (literally just shoot them, wait, or dragnet them - some combo thereof - two of which might work here for something a lot of antags get for free.)
> 
> Honestly, I'd feel a lot better about this if it was just a two-time stimpack autoinjector that triggers on hitting softcrit or manually.
> 
> This is just a crazy buff across the board around how people in Syndie mods usually go down, but it won't make everyone jump the hardsuit guy in red - it'll make everyone who has to deal with hardsuit guy as a restricted antag on MRP think they do and shoot him in the middle of a soliloquy to steal his drip (basically treating him more like a desword nukie even if he's not.)
> 
> This ain't all bad, because fictionally - wearing the stuff and having it be scary is cool if that's what you're after, and why would you wear the tech of the terrible Cybersun/Gorlex killsquads if you didn't absolutely have to?
> 
> I'm just concerned at how much of a swing on balance this is going to be - baton knockdown is unfortunately a cornerstone fallback even as we chip away at stun combat.

This module only prevents baton knockdown, it does nothing against bodythrows and shovestuns.

The entire point of nukies is that they are meant to be powerful, them dying on a single baton hit, kinda goes against that philosophy,not every nukie runs stims and energy shield module.

You shouldn't be able to fight them fair and square and win, and i'd argue even a traitor should have an advantage against a single security officer, since more often than now you are fighting them in pairs.

The base mod is no longer obtainable in space, or lavaland, unless a miner raids the syndie base, which is a rare spawn.

It's not that big of a buff, this is merely to prevent fights to be over in 1 click of a baton, and if it ends up shaking the meta, GOOD, maybe this will open up some buffs to  sec other tools now that the stunbatons aren't the universal answer for everyhing.

**Comment by SomeoneYouProbablyKnow** [comment]
> I can't remotely stress how tweaked this will be in the hands of ops and anyone with access to a little bit of extra prep. Free knockdown immunity, in no-grav/jetpack settings especially, is going to be ridiculous for how hard it makes to disarm or contest people

Good, a fight between a geared up space terrorist and an overworked surgeon with a stick should not be even.

I do think you preventing the whole knockdown leads to some problems with not being able to escape from a mod-suited attacker. Getting hit by a baton even with this module should slow you down for a moment or two.

> The base mod is no longer obtainable in space, or lavaland, unless a miner raids the syndie base, which is a rare spawn.

What's more likely is sec stealing an uplink and getting this module, drastically reducing what station side antags can do against them which is a big issue and should be adressed.

> Honestly, I'd feel a lot better about this if it was just a two-time stimpack autoinjector that triggers on hitting softcrit or manually.

Also yea some resource management aspect sounds good. Even ninja modsuit anti-stun thing needs to be recharged. Could just be a hefty power requirements. Upgrade your batteries before tanking a baton

**Comment by EnterTheJake** [comment]
> > I can't remotely stress how tweaked this will be in the hands of ops and anyone with access to a little bit of extra prep. Free knockdown immunity, in no-grav/jetpack settings especially, is going to be ridiculous for how hard it makes to disarm or contest people
> 
> Good, a fight between a geared up space terrorist and an overworked surgeon with a stick should not be even.
> 
> I do think you preventing the whole knockdown leads to some problems with not being able to escape from a mod-suited attacker. Getting hit by a baton even with this module should slow you down for a moment or two.
> 
> > The base mod is no longer obtainable in space, or lavaland, unless a miner raids the syndie base, which is a rare spawn.
> 
> What's more likely is sec stealing an uplink and getting this module, drastically reducing what station side antags can do against them which is a big issue and should be adressed.
> 
> > Honestly, I'd feel a lot better about this if it was just a two-time stimpack autoinjector that triggers on hitting softcrit or manually.
> 
> Also yea some resource management aspect sounds good. Even ninja modsuit anti-stun thing needs to be recharged. Could just be a hefty power requirements. Upgrade your batteries before tanking a baton

Baton resistance does nothing to the slowdown from stamina damage.

**Comment by SomeoneYouProbablyKnow** [comment]
Yes but a person who was knocked down has to get up and then has to deal with the slowdown. Remove the knockdown and you give them essentially extra speed in the form of not having to crawl and get up. Eating a knockdown should slow you down even more than the stamina damage

**Comment by EnterTheJake** [comment]
> Yes but a person who was knocked down has to get up and then has to deal with the slowdown. Remove the knockdown and you give them essentially extra speed in the form of not having to crawl and get up. Eating a knockdown should slow you down even more than the stamina damage

no you do not, the slowdown is tied to loss of stamina not the knockdown. you can test it yourself, give yourself the trait TRAIT_BATON_RESISTANCE and you will start moving considerably slower after you tank a baton hit.

**Comment by SomeoneYouProbablyKnow** [comment]
Okay... imagine a race between a modsuited person and a non modsuited person. They both get hit by a baton, one goes down the other keeps running with the slowdown. 
During the time it took for the non modsuited person to get up and starts running again the modsuited one already gained a substantial lead not due to any coded slowdown but due to not dropping to the ground in the first place. 
If i'm chasing someone with a baton, and i get hit, i have better chances of catching up WITH this module. It's not only useful in combat it's also useful in chasing down weaker opponents which are different situations. It's okay for combat to be unfair as long as you can feasibility run the fuck away.

**Comment by totalglyphiddeath** [comment]
Extremely rare based balance PR.

**Comment by Livrah** [comment]
almost like the "knockdown of the floor" fix was a mistake

**Comment by necromanceranne** [comment]
> if we have to add baton resistance to every single antag then i don't think the lack of baton resistance is the problem

ah, if only I had had my way...

**Comment by Sakamoto4ka** [comment]
imagine using a stunstick designed for non-lethal detention against a terrorist because it is the most effective weapon for lethal detention (you need to hit the sprite 2 times)

**Comment by Metekillot** [comment]
I don't think it's a good idea to give unilateral baton resistance to a mod suit module that is an exclusive to that antagonist using it. 

I think a better idea would be to just give nukies the option to buy CNS rebooters. Maybe traitors, too, at a high enough threat level.

The fact that everyone also seems to ignore the multiple reagents that we have that also give you baton resistance also frustrates me a bit; it gets me to feeling that people don't care about balance so much as building a narrative.

**Comment by necromanceranne** [comment]
Nuclear Ops can already buy the CNS Rebooter. Spies sometimes get it too.

**Comment by Formerxboxgamer** [comment]
It’s okay for those people who are newer to falter on their role. I don’t expect them to not mess up, but I do expect them to learn as they go and eventually become the scary nukie or traitor they can be. If we feed everyone the perfect experience on day one, it cheapens the accomplishment of finally earning victory while also making things miserable for people on the other side of the experience because they feel like they can’t fight back or do anything.

not having experience or knowledge is fine, everyone flukes a few rounds before getting the hang of it, and that’s fine too.

**Comment by necromanceranne** [comment]
Well, I personally don't think it is fair on them and feels like a failure on our part to design the game if their only way to get better is to be thrown into the deep end. It certainly makes life easier on the already experienced to have a lot of less experienced players to dunk on, but it isn't terribly encouraging for those players on the receiving end. They are our majority of players, and so they should be our focus. Not the veteran players.

**Comment by Formerxboxgamer** [comment]
I’m fine with making things more approachable, but turning an antag into “Ok now it’s your turn to murderbone” isn’t engaging enough gameplay to keep those new players around and wouldn’t make the game fun for the other new players who didn’t roll nukie. We should be encouraging people to learn through doing rather than saying “Nah you don’t need to learn, you rolled the kill everyone role”

**Comment by jughu** [comment]
i think the traitor version should have it but exclude it from the nukie version

**Comment by Loafin34** [comment]
what if we had shin guards that made us invincible to telebatong and policebatong knockdown so that you can't cheese an arrest by hittting someone with a telebaton

**Comment by MatrixOne500** [comment]
Terrible change that only benefits murderboning. Sec should be rewarded for being able to get into melee range to apply the baton. Actually bad change for the game that'll only ruin shifts, cause them to end early, help murderboners, and make sec even more hopeless against hyperantags.

**Comment by xzero314** [comment]
I think this should be an item in the Nuke op uplink for nuke ops to get and add to their suit.

**Comment by SomeoneYouProbablyKnow** [comment]
Isn't this what the rebooter implant for? Uplink's just got so much bloat people can't find the useful items

**Comment by EnterTheJake** [comment]
> I think this should be an item in the Nuke op uplink for nuke ops to get and add to their suit. I don't think it should be something baseline with all syndicate mods.

That would defeat the point of making nuke ops or valid traitors more accessible to new players.

**Comment by EnterTheJake** [comment]
> Isn't this what the rebooter implant for? Uplink's just got so much bloat people can't find the useful items

As the guy who reworked the rebooter implant, i can probably chip in on this

Rebooter is supposed to be a hard expensive counter to stamina damage, it's 8 tc + 5 for the advanced EMP shield if you don't want it to backfire.

This change is supposed as Anne elegantly put out to make nuke ops more threatening and more forgiving for new players

Nuclear operatives are supposed to be powerful, a nukie getting immediately killed by getting clicked once, completely ruins that.

**Comment by SomeoneYouProbablyKnow** [comment]
> Nuclear operatives are supposed to be powerful, a nukie getting immediately killed by getting clicked once, completely ruins that.

I agree, but as long as stun combat exists it's gonna be a tightrope to balance. You don't want nuckies to be overwhelmingly powerful, you should be able to at least run away. Else why even bother if you have no agency in the situation. If you're not careful you'll just turn the dynamic around where it's completely unforgiving towards new players facing nuckies.
That's why i hate things like jetpack combat, not only is it literally sickening but it also cuts off your means of escape due to the gigantic speed difference.

Also you haven't adressed my concern about this being INCREDIBLY valuable to secs who manage to yoink an uplink (a common occurence). Crippling any crewside traitor relying on a baton.

**Comment by EnterTheJake** [comment]
> > Nuclear operatives are supposed to be powerful, a nukie getting immediately killed by getting clicked once, completely ruins that.
> 
> I agree, but as long as stun combat exists it's gonna be a tightrope to balance. You don't want nuckies to be overwhelmingly powerful, you should be able to at least run away. Else why even bother if you have no agency in the situation. If you're not careful you'll just turn the dynamic around where it's completely unforgiving towards new players facing nuckies. That's why i hate things like jetpack combat, not only is it literally sickening but it also cuts off your means of escape due to the gigantic speed difference.
> 
> Also you haven't adressed my concern about this being INCREDIBLY valuable to secs who manage to yoink an uplink (a common occurence). Crippling any crewside traitor relying on a baton.

Thing is, they are not, last time i checked war ops sit around 30%, if this ends up increasing nukie winrate, i'd say it's a good thing.

As for this being given to sec, it's out of my hands, i wanted it to be unremovable but Fikou wants otherwise.

**Comment by SomeoneYouProbablyKnow** [comment]
Unironically who cares about nuckie winrate. It's about the individual fights, i'd rather nuckies always lose if the fights they get into feel fun and survivable instead of a perfect 50% winrate based completely on nuckies steamrolling newbies untill they get to Space McSpacy with 6000 hours who zig zags and shoots them with a laser gun till they die. (Both are hyperbole just so we are clear)
Also that's a whole different discussion but considering our main mode is dynamic and nuckies nuke the station if they win 50% arguably shouldn't be the goal as nuckies are not the only thing in the round.
Also also, isn't war ops winrate like quite a bit lower than regular nuckie winrate? I'd have to check but fuck war ops.

As for sec, the removeable part is not the issue, it's the fact that it's purchasable by traitors. No one is supposed to get loot of a nuckie in the first place but uplink theft is rampant, maybe limit it to 1 purchase per uplink, i don't think you can get away with a more sophisticated deterrent sadly.

**Comment by zoomachina** [comment]
> Unironically who cares about nuckie winrate. It's about the individual fights, i'd rather nuckies always lose if the fights they get into feel fun and survivable instead of a perfect 50% winrate based completely on nuckies steamrolling newbies untill they get to Space McSpacy with 6000 hours who zig zags and shoots them with a laser gun till they die. (Both are hyperbole just so we are clear) 

"Nukie winrate should rather be 0% than depend on nukie/station skill"

**Comment by Metekillot** [comment]
> > Isn't this what the rebooter implant for? Uplink's just got so much bloat people can't find the useful items
> 
> As the guy who reworked the rebooter implant, i can probably chip in on this
> 
> Rebooter is supposed to be a hard expensive counter to stamina damage, it's 8 tc + 5 for the advanced EMP shield if you don't want it to backfire.
> 
> This change is supposed as Anne elegantly put out to make nuke ops more threatening and more forgiving for new players
> 
> Nuclear operatives are supposed to be powerful, a nukie getting immediately killed by getting clicked once, completely ruins that.
> 
> 

I respect wanting to make the game more accessible to new players but at a certain point being a new player means that you're going to get your ass kicked even if you are handed the most powerful equipment in the game. That's an inevitable learning experience. 

And if we are taking the tack of making it more accessible to new players than I would say it might be pertinent for all the people trying to inject the knockdown exploit fix disarm thing into the discussion to maybe make a different PR about it. 

And increasing nukie win rate while making them more threatening:

A lot of nuclear operative players would benefit quite a lot from deciding to do any kit at all besides energy shield+DSword because from their point of view the absorbed attacks and blocked attacks means that they're more survivable. 


I see almost none of them use the stimulants that do literally actually make you almost totally immune to stuns while passively healing you a shitload. 


I hardly see any of them using their combat stealth options or their ambush options. 

It's always the energy shield and D-Sword guys who yell the loudest about getting hit in melee combat while using their melee kit that doesn't compensate for stuns whatsoever.

**Comment by Metekillot** [comment]
image

Seriously, they all squawk about how immensely overpowered stun batons are but you have this in your pocket for the price of 5 TC. Then there's maintenance pump-up, ephedrine, bath salts, and meth, not to mention the CNS rebooter and the anti-drop implants.

I'd say a good balance change is having the stimpack injection be adjustable but the rest of this argument about stun batons just seems like a nothingburger.

**Comment by EnterTheJake** [comment]
> 

Stims are pretty ass, tho, they grant you a meager 5 stamina regen per tick, not even enough to tank 2 batons and they are one time use for 5 tc.

**Comment by SomeoneYouProbablyKnow** [comment]
Stims are pretty good if you know how to use em but if the goal's to help new players stims ain't it. You need to use them at the right time, know how to use the extra speed, not overextend, click at high speeds etc.
Frankly the rebooter is the best newbie item, why not just add it to some of the melee kits or even starting nuckie gear

**Comment by EnterTheJake** [comment]
> Stims are pretty good if you know how to use em but if the goal's to help new players stims ain't it. You need to use them at the right time, know how to use the extra speed, not overextend, click at high speeds etc. Frankly the rebooter is the best newbie item, why not just add it to some of the melee kits or even starting nuckie gear

Stims are a must have because they are the only item in all of the syndicate arsenal that has baton resist, nobody would use em otherwise.

**Comment by Metekillot** [comment]
> > 
> 
> Stims are pretty ass, tho, they grant you a meager 5 stamina regen per tick, not even enough to tank 2 batons and they are one time use for 5 tc.

Did you not read the rest of the description on what effects stimulants have?

**Comment by Loafin34** [comment]
we should probably put in the stimulant's description how it completely invalidates batons so that this isn't miscommunicated

**Comment by Metekillot** [comment]
The exact statistics of stamina are obfuscated so I don't fault people for not understanding how immensely powerful stimulants are.

**Comment by Metekillot** [comment]
I totally withdraw what I recommended about using syndie stims to counter baton stam crit. These things kinda suck ass for fighting batons??

post-donut runspeed for god's weakest sec officer

https://github.com/tgstation/tgstation/assets/49173900/8e1b0bf0-12bd-4bac-8d2c-f0381508eb34

post-baton runspeed for god's strongest stimulant medipen user

https://github.com/tgstation/tgstation/assets/49173900/4e94a948-1686-465c-8b1f-14c2ef174860

**Comment by zoomachina** [comment]
> I totally withdraw what I recommended about using syndie stims to counter baton stam crit. These things kinda suck ass for fighting batons??
> 
> post-donut runspeed for god's weakest sec officer
> 
>  normal.mp4 
> post-baton runspeed for god's strongest stimulant medipen user
> 
>  stunned.mp4

now imagine the latter but without the speed-up from stimulants: that would be the speed of an antag with this modsuit module (possibly less because modsuits can slow you down)

**Comment by EnterTheJake** [comment]
> I totally withdraw what I recommended about using syndie stims to counter baton stam crit. These things kinda suck ass for fighting batons??

ooooh how the turn tables.....

**Comment by DrAmazing343** [comment]
Thank you for posting that. People were telling me I was wrong and out of touch for saying that Syndie Stims do basically nothing, but I think it's true— there's very little difference between Maintenance Pump-Up and Syndie Stims that I can see, except that one's 5TC and the other is maints loot.

**Comment by EnterTheJake** [comment]
> if you want to. you can also add it to black market and whatever, i just dont like the recent trend of invisible module that affects combat where you just need to know that this specific suit has one- it bites on the modularity of the system when everything is inherent the sprite can be whatever you think would work to visually convey it, some sort of pad on the shoulders, thing attached to belt, etc

Ok it has a visual effect, complexity and cell drain added to it.

I also removed it from the infiltrator suit like you requested and added it to the uplink.

File: code/modules/mod/modules/modules_general.dm [hunk 1]
```diff
@@ -979,3 +979,30 @@
 	attempt_insert_storage(product)
 	balloon_alert(mod.wearer, "ammo box dispensed.")
 	playsound(src, 'sound/machines/microwave/microwave-end.ogg', 50, TRUE)
+
+/obj/item/mod/module/shock_absorber
+	name = "MOD shock absorption module"
+	desc = "A module that makes the user resistant to the knockdown inflicted by Stun Batons."
+	icon_state = "no_baton"
+	complexity = 1
+	idle_power_cost = 0
```
Comment (Fikou) [inline code/modules/mod/modules/modules_general.dm:988]: do they have a power cost by default? if not then you can remove this line

File: code/modules/mod/modules/modules_general.dm [hunk 2]
```diff
@@ -979,3 +979,30 @@
...
+
+/obj/item/mod/module/shock_absorber/on_suit_deactivation(deleting)
+	. = ..()
+	REMOVE_TRAIT(mod.wearer, TRAIT_BATON_RESISTANCE, REF(src))
+	UnregisterSignal(mod.wearer, COMSIG_MOB_BATONED)
+
+/obj/item/mod/module/shock_absorber/proc/do_visuals(datum/source)
+	SIGNAL_HANDLER
+	drain_power(use_energy_cost)
+	var/datum/effect_system/lightning_spread/s = new /datum/effect_system/lightning_spread
```
Comment (Fikou) [inline code/modules/mod/modules/modules_general.dm:1006]: [you cannot use singlehttps://github.com/tgstation/tgstation/blob/master/.github/guides/STYLE.md#use-descriptive-and-obvious-names](https://github.com/tgstation/tgstation/blob/master/.github/guides/STYLE.md#use-descriptive-and-obvious-names)

Comment (Fikou) [inline code/modules/mod/modules/modules_general.dm:1006]: https://github.com/tgstation/tgstation/blob/master/.github/guides/STYLE.md#use-descriptive-and-obvious-names

File: code/modules/uplink/uplink_items/suits.dm [hunk 3]
```diff
@@ -64,6 +64,12 @@
 	item = /obj/item/mod/module/noslip
 	cost = 2
 
+/datum/uplink_item/suits/shock_absorber
+	name = "MODsuit Shock-Absorber Module"
+	desc = "A MODsuit module preventing the user from getting knocked down by batons"
```
Comment (Fikou) [inline code/modules/uplink/uplink_items/suits.dm:69]:
```suggestion
	desc = "A MODsuit module preventing the user from getting knocked down by batons."
```

File: code/modules/uplink/uplink_items/suits.dm [hunk 4]
```diff
@@ -64,6 +64,12 @@
...
+	item = /obj/item/mod/module/shock_absorber
+	cost = 4
```
Comment (Fikou) [inline code/modules/uplink/uplink_items/suits.dm:71]: maybe drop this to 2-3. you already have to spend tc on the suit itself and noslips are 2

**Comment by Jacquerel** [comment]
> if you want to. you can also add it to black market and whatever, i just dont like the recent trend of invisible module that affects combat where you just need to know that this specific suit has one- it bites on the modularity of the system when everything is inherent the sprite can be whatever you think would work to visually convey it, some sort of pad on the shoulders, thing attached to belt, etc

Seems this still doesn't have an equipped sprite, is this a merge requirement? I think it does at the very least sound like a good idea.

**Comment by EnterTheJake** [comment]
> > if you want to. you can also add it to black market and whatever, i just dont like the recent trend of invisible module that affects combat where you just need to know that this specific suit has one- it bites on the modularity of the system when everything is inherent the sprite can be whatever you think would work to visually convey it, some sort of pad on the shoulders, thing attached to belt, etc
> 
> Seems this still doesn't have an equipped sprite, is this a merge requirement? I think it does at the very least sound like a good idea.

Me and Fikou had an agreement that adding some sort of visual effects whenever you get batoned was the better solution.

So i just added the Mjolnir sparks to the module whenever a baton hit occurs.

https://www.youtube.com/watch?v=3_BKjYH5uzY.

Adding a sprite on top of the already armored nukie suits would look pretty cluttered.

File: code/modules/mod/modules/modules_general.dm [hunk 5]
```diff
@@ -979,3 +979,29 @@
...
+/obj/item/mod/module/shock_absorber/on_suit_deactivation(deleting)
+	. = ..()
+	REMOVE_TRAIT(mod.wearer, TRAIT_BATON_RESISTANCE, REF(src))
+	UnregisterSignal(mod.wearer, COMSIG_MOB_BATONED)
+
+/obj/item/mod/module/shock_absorber/proc/mob_batoned(datum/source)
+	SIGNAL_HANDLER
+	drain_power(use_energy_cost)
+	var/datum/effect_system/lightning_spread/sparks = new /datum/effect_system/lightning_spread
+	sparks.set_up(5, 1, mod.wearer.loc)
```
Comment (Fikou) [inline code/modules/mod/modules/modules_general.dm:1006]: use named arguments if possible in this proc

File: code/modules/mod/mod_types.dm [hunk 6]
```diff
@@ -220,6 +220,7 @@
 	starting_frequency = MODLINK_FREQ_SYNDICATE
 	applied_cell = /obj/item/stock_parts/cell/super
 	applied_modules = list(
+		/obj/item/mod/module/shock_absorber,
```
Comment (Fikou) [inline code/modules/mod/mod_types.dm:223]: should go below the storage module in each one, just a nitpick

File: code/game/objects/items/melee/baton.dm [hunk 7]
```diff
@@ -212,6 +212,7 @@
 		if(!trait_check)
 			target.Knockdown((isnull(stun_override) ? knockdown_time : stun_override))
 		additional_effects_non_cyborg(target, user)
+	SEND_SIGNAL(target, COMSIG_MOB_BATONED)
```
Comment (Fikou) [inline code/game/objects/items/melee/baton.dm:215]: add some more arguments for people that could use it in the future, like the baton used and the attacker

**Review by Fikou — CHANGES_REQUESTED** [review]

File: code/game/objects/items/melee/baton.dm (hunk 7 above)
Comment (Xander3359) [reply code/game/objects/items/melee/baton.dm:215]: Wouldn't that just be simple for someone to add in the future if they were to ever need it?

**Review by Fikou — APPROVED** [review]
code looks good

**Comment by Jackal-boop** [comment]
If this gets merged I'd be interested to see how we could change the operative mod shield to be less of a straight buff.

**Comment by EnterTheJake** [comment]
> If this gets merged I'd be interested to see how we could change the operative mod shield to be less of a straight buff.

Energy Shield mod already got nerfed, sometimes simple passive buffs are a good thing, Energy shield mod exists to prevent cheap oneshots.

**Comment by adamkad1** [comment]
I'd rather get a bomb resistance module

**Pull request merged**